### DISCUSSION TITLE
Set my custom id for uploaded file

### DISCUSSION THREAD

**Discussion by pas8690** [directus/directus#15000] [Archive]
Hi guys, 
i've a question regarding the opportunity to set my personal id on file object during files upload. 
Is this behavior possible  ? 

An ideal example would be

```
url: api_base_url + '/files',
  file: {
    "path": "blob:https://platform.preview.appgyver.com/cc59fe7a-0fa3-4c4c-9343-656c2be2e6a1",
    "name": "image.jpeg",
    "size": 131790,
    "mimeType": "image/jpeg"
    **"id" : "MY-CUSTOM-ID-GOES-HERE"**
  },
  authorization: 'Bearer xxxxxx'
```

Thank you and have a nice day

**Comment by azrikahar** [comment]
You should be able to pass the id with a UUID v4 value in the form data _before_ the file part. You can also do the same with any other fields except `filename_disk` and `filename_download`. 

Example:

![chrome_ctJPy4qzTk](https://user-images.githubusercontent.com/42867097/183882798-0a72a721-d08e-4ff9-8e1e-f0de1e03f2cc.png)

Further reference in the docs regarding this: https://docs.directus.io/reference/files/#upload-a-file
